File: frontend/src/app/data/RecentGameDto.ts
export class RecentGameDto {
  uuid: string;
  playerName: string;
  playerCount: number;
  victoryPoints: number;
  turns: number;
  date: string;
  isCrisis: boolean;

  constructor(uuid: string,
              playerName: string,
              playerCount: number,
              victoryPoints: number,
              turns: number,
              date: Date,
              isCrisis: boolean) {
    this.uuid = uuid;
    this.playerName = playerName;
    this.playerCount = playerCount;
    this.victoryPoints = victoryPoints;
    this.turns = turns;
    this.date = date ? RecentGameDto.formatDate(date) : '';
    this.isCrisis = isCrisis;
  }

  private static formatDate(date: Date): string {
    const dateObject = new Date(date);

    return dateObject.toLocaleString('en-US', {
      year: 'numeric',
      month: 'short',
      day: 'numeric',
      hour: '2-digit'
    });
  }
}

File: src/main/java/com/terraforming/ares/dto/RecentGameDto.java
package com.terraforming.ares.dto;

import jakarta.persistence.Id;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

/**
 * Created by oleksii.nikitin
 * Creation date 17.05.2022
 */
@Getter
@Setter
@Builder
public class RecentGameDto {

    @Id
    private String uuid;
    private String playerName;
    private int playerCount;
    private int victoryPoints;
    private int turns;
    private String date;
    private boolean isCrisis;

}
